pkg/util/atomicfs/fsync.go

```diff
@@ -68,22 +68,7 @@ func (a *File) Close() error {
 	}
 
 	cleanup = false
-	// After writing the file and calling fsync on it, fsync the containing directory
-	// to ensure the directory entry is persisted to disk.
-	//
-	// From https://man7.org/linux/man-pages/man2/fsync.2.html
-	// > Calling fsync() does not necessarily ensure that the entry in the
-	// > directory containing the file has also reached disk.  For that an
-	// > explicit fsync() on a file descriptor for the directory is also
-	// > needed.
-	dir, err := os.Open(filepath.Dir(a.finalPath))
-	if err != nil {
-		return err
-	}
-
-	merr.Add(dir.Sync())
-	merr.Add(dir.Close())
-	return merr.Err()
+	return syncDir(filepath.Dir(a.finalPath))
```

Comment:
Changes in `atomicfs` and `shutdownmarker` packages come from previous design, when we planned to store "downscale marker" on the disk. However we have moved on from that design, and now changes in these two packages are unrelated to this PR.

I suggest we remove changes to `atomicfs` and `shutdownmarker` packages from this PR.

Reply:
Done in https://github.com/grafana/mimir/pull/8956/commits/0d64c249f3461cce83c8f3a730fdbdb0abe1ef8e.